### PR TITLE
Groups

### PR DESCRIPTION
**Pull request by AadSchippers**
This feature adds some functionality to group maintanance.

Changes:
- Only admins or group owners can update or delete a group;
- Anyone who is logged in can join any group. For open groups (isOpen = true) default membership is 'member', for closed groups (isOpen = false) default membership is 'pending';
- Members can be added to a group, by group owner or admin, using ChangeMembershipGroup;
- Membership can be altered, by group owner or admin, using ChangeMembershipGroup;
- Members can be removed from a group, by group owner or admin, using RemoveMembershipGroup;
- Added ability to refuse group membership when 2FA is not enabled;
- CreateGroup: creator is automatically set as owner of newly created group;
- Fixed returning proper comments;
- Added logging.

### CODE REVIEW

File: core/mutation.py [hunk 1]
```diff
@@ -134,17 +146,21 @@ def mutate(self, info, id, input):
         try:
             with reversion.create_revision():
                 group = Group.objects.get(pk=get_id(id))
+                if not group.can_change(info.context.user):
```
Comment (bartjkdp) [inline core/mutation.py]: would be nice to put this above `revision.create_revision()`

File: core/mutation.py [hunk 2]
```diff
@@ -160,13 +176,23 @@ def mutate(self, info, id):
         try:
             with reversion.create_revision():
                 group = Group.objects.get(pk=get_id(id))
-                group.delete()
+                if not group.can_change(info.context.user):
```
Comment (bartjkdp) [inline core/mutation.py]: idem

**Comment by bartjkdp** [comment]
Looks good to me!

File: core/auth.py [hunk 3]
```diff
@@ -18,6 +18,9 @@ def create_user(self, claims):
             name=claims.get('name'),
             email=claims.get('email'),
             picture=claims.get('picture'),
+            is_government = claims.get('is_government'),
+            has_2FA_enabled = claims.get('has_2FA_enabled'),
```
[the remark this answers is not in the document]
Comment (AadSchippers) [reply core/auth.py]: Done.

File: core/auth.py [hunk 4]
```diff
@@ -27,6 +30,8 @@ def update_user(self, user, claims):
             user.name = claims.get('name')
             user.email = claims.get('email')
             user.picture = claims.get('picture')
+            user.is_government = claims.get('is_government')
+            user.has_2FA_enabled = claims.get('has_2FA_enabled')
```
[the remark this answers is not in the document]
Comment (AadSchippers) [reply core/auth.py]: Done.

File: core/mutation.py (hunk 1 above)
Comment (AadSchippers) [reply core/mutation.py]: Done.

File: core/mutation.py (hunk 2 above)
Comment (AadSchippers) [reply core/mutation.py]: Done.

**Pull request merged**